src/middleware/traceUser.ts

```diff
@@ -0,0 +1,36 @@
+import { NextFunction, Request, Response } from "express"
+import { Session } from "express-session"
+
+import tracer from "@utils/tracer"
+
+import { SessionData } from "@root/types/express/session"
+
+type UserSession = Session & SessionData
```

Comment:
seems quite similar to UserWithSiteSessionData or RequestWithSession, could we just use that type instead?

Reply:
Yeah, I was hoping to reuse some existing types too, but I wasn't satisfied with the existing types 😞
* `UserWithSiteSessionData` requires a sitename, which may not be there, and is not needed by this middleware (although I suppose if it is present, we could shove it in the trace too 🤔 )
* `RequestWithSession` extends `RequestWithGrowthBook` which also felt a bit weird wrong (not sure?) . growthbook is not needed in this middleware (at the moment anyway).